examples/bri-3/src/bri/identity/bpiAccounts/api/accounts.controller.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { BadRequestException, NotFoundException } from '@nestjs/common';
+import { CqrsModule } from '@nestjs/cqrs';
+import { Test, TestingModule } from '@nestjs/testing';
+
+import { NOT_FOUND_ERR_MESSAGE } from './err.messages';
+import { NOT_FOUND_ERR_MESSAGE as SUBJECT_ACCOUNT_NOT_FOUND_ERR_MESSAGE } from '../../bpiSubjectAccounts/api/err.messages';
+import { AccountController } from './accounts.controller';
+import Mapper from '../../../utils/mapper';
+import { BpiAccountStorageAgent } from '../agents/bpiAccountsStorage.agent';
+import { BpiAccountAgent } from '../agents/bpiAccounts.agent';
+import { MockBpiAccountsStorageAgent } from '../agents/mockBpiAccountStorage.agent';
+import { SubjectModule } from '../../bpiSubjects/subjects.module';
+import { CreateBpiAccountCommandHandler } from '../capabilities/createBpiAccount/createBpiAccountCommand.handler';
+import { DeleteBpiAccountCommandHandler } from '../capabilities/deleteBpiAccount/deleteBpiAccountCommand.handler';
+import { GetAllBpiAccountsQueryHandler } from '../capabilities/getAllBpiAccounts/getAllBpiAccountQuery.handler';
+import { GetBpiAccountByIdQueryHandler } from '../capabilities/getBpiAccountById/getBpiAccountByIdQuery.handler';
+import { UpdateBpiAccountCommandHandler } from '../capabilities/updateBpiAccount/updateBpiAccountCommand.handler';
+import { BpiSubjectAccountAgent } from '../../bpiSubjectAccounts/agents/bpiSubjectAccounts.agent';
+import { BpiSubjectAccountStorageAgent } from '../../bpiSubjectAccounts/agents/bpiSubjectAccountsStorage.agent';
+import { MockBpiSubjectAccountsStorageAgent } from '../../bpiSubjectAccounts/agents/mockBpiSubjectAccountsStorage.agent';
+import { CreateBpiAccountDto } from './dtos/request/createBpiAccount.dto';
+import { SubjectController } from '../../bpiSubjects/api/subjects.controller';
+import { SubjectAccountController } from '../../bpiSubjectAccounts/api/subjectAccounts.controller';
+import { CreateBpiSubjectDto } from '../../bpiSubjects/api/dtos/request/createBpiSubject.dto';
+import { CreateBpiSubjectAccountDto } from '../../bpiSubjectAccounts/api/dtos/request/createBpiSubjectAccount.dto';
+import { SubjectAccountModule } from '../../bpiSubjectAccounts/subjectAccounts.module';
+import { validate as uuidValidate, version as uuidVersion } from 'uuid';
+import { BpiSubjectStorageAgent } from '../../bpiSubjects/agents/bpiSubjectsStorage.agent';
+import { MockBpiSubjectStorageAgent } from '../../bpiSubjects/agents/mockBpiSubjectStorage.agent';
+
+describe.only('AccountController', () => {
+  let accountController: AccountController;
```

Comment:
I used the approach where i registered only the mock storage from other modules and creating the necessary data directly in it, to avoid going through the separate controller as it is already tested in it's own test file. You can see an example in messages spec file. Not sure which approach is better but we should definitely stick to one of them. I am lightly inclined to my approach as the unit tests are then limited only to the scope of the single controller.

Reply:
and you need less stuff to initialize the test module

Reply:
ok i'll change